sparkplug-core/test/sparkplug/kryo_test.clj

```diff
@@ -1,10 +1,53 @@
 (ns sparkplug.kryo-test
   (:require
-    [clojure.test :refer [deftest testing is]]
-    [sparkplug.kryo :as kryo]))
+    [clojure.test :refer [are deftest testing is]]
+    [clojure.test.check.clojure-test :refer [defspec]]
+    [clojure.test.check.generators :as gen]
+    [clojure.test.check.properties :as prop]
+    [sparkplug.kryo :as kryo])
+  (:import
+    com.esotericsoftware.kryo.Kryo
+    (com.esotericsoftware.kryo.io
+      Input
+      Output)))
 
 
 (deftest classpath-search
   (let [registries (kryo/classpath-registries)]
     (is (sequential? registries))
     (is (<= 2 (count registries)))))
+
+
+(defn- serialize
+  ^bytes [^Kryo kryo obj]
+  (let [output (Output. 4096 -1)]
+    (.writeClassAndObject kryo output obj)
+    (.flush output)
+    (.getBuffer output)))
+
+
+(defn- deserialize
+  [^Kryo kryo ^bytes data]
+  (let [input (Input. data)]
+    (.readClassAndObject kryo input)))
+
+
+(defn- replace-nans
```

Comment:
Note that this does not entirely cover your bases if `gen/any` produces regular expression patterns, since those (bewilderingly) don't compare equal in Java.